src/main/java/competition/subsystems/pose/PoseSubsystem.java

```diff
@@ -13,11 +17,22 @@
 
     final DriveSubsystem drive;
 
+    // odometry
+    final DifferentialDriveOdometry odometry;
+    final DoubleProperty robotX;
+    final DoubleProperty robotY;
+    final DoubleProperty robotHeading;
+
     @Inject
     public PoseSubsystem(CommonLibFactory clf, PropertyFactory propManager, DriveSubsystem drive) {
         super(clf, propManager);
-
         this.drive = drive;
+
+        odometry = new DifferentialDriveOdometry(getRotation2DGyroAngle());
+        robotX = propManager.createEphemeralProperty("robotX", 0.0);
+        robotY = propManager.createEphemeralProperty("robotY", 0.0);
+        robotHeading = propManager.createEphemeralProperty("robotHeading", 0.0);   
+        odometry.resetPosition(odometry.getPoseMeters(), getRotation2DGyroAngle());
```

Comment:
What is this doing? Are you resetting the odometry's position to it's own position?

Reply:
resetPosition resets the robot's position on the field.

Documentations:

`  /**
   * Resets the robot's position on the field.
   *
   * <p>You NEED to reset your encoders (to zero) when calling this method.
   *
   * <p>The gyroscope angle does not need to be reset here on the user's robot code.
   * The library automatically takes care of offsetting the gyro angle.
   *
   * @param poseMeters The position on the field that your robot is at.
   * @param gyroAngle  The angle reported by the gyroscope.
   */
  public void resetPosition(Pose2d poseMeters, Rotation2d gyroAngle) {
    m_poseMeters = poseMeters;
    m_previousAngle = poseMeters.getRotation();
    m_gyroOffset = m_poseMeters.getRotation().minus(gyroAngle);

    m_prevLeftDistance = 0.0;
    m_prevRightDistance = 0.0;
  }`